packages/js/src/plugins/auctionHouseModule/operations/buy.ts

```diff
@@ -0,0 +1,214 @@
+import { ConfirmOptions, PublicKey } from '@solana/web3.js';
+import type { Metaplex } from '@/Metaplex';
+import { Option } from '@/utils';
+import {
+  Operation,
+  OperationHandler,
+  Pda,
+  Signer,
+  SolAmount,
+  SplTokenAmount,
+  useOperation,
+} from '@/types';
+import { SendAndConfirmTransactionResponse } from '../../rpcModule';
+import { AuctionHouse, Listing, Purchase } from '../models';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'BuyOperation' as const;
+
+/**
+ * Creates a bid on a given asset and then executes a sale on the created bid and listing.
+ *
+ * ```ts
+ * await metaplex
+ *   .auctionHouse()
+ *   .buy({ auctionHouse, listing, buyer })
+ *   .run();
+ * ```
+ *
+ * @group Operations
+ * @category Constructors
+ */
+export const buyOperation = useOperation<BuyOperation>(Key);
+
+/**
+ * @group Operations
+ * @category Types
+ */
+export type BuyOperation = Operation<typeof Key, BuyInput, BuyOutput>;
+
+/**
+ * @group Operations
+ * @category Inputs
+ */
+export type BuyInput = {
+  /** The Auction House in which to create a Bid and execute a Sale. */
+  auctionHouse: AuctionHouse;
```

Comment:
Minimum Valuable Input!

Reply:
Why do we need the full object for `executeSale`?

Reply:
Ignore me, I know why. We need it to recreate the LazyPurchase.

Reply:
I am not sure, but it is required by `ExecuteSaleInput` to have `auctionHouse: AuctionHouse;`